templates/go/message.go

```diff
@@ -1,13 +1,26 @@
 package golang
 
+// Embedded message validation.
 const messageTpl = `
 	{{ $f := .Field }}{{ $r := .Rules }}
 	{{ template "required" . }}
 
 	{{ if .MessageRules.GetSkip }}
 		// skipping validation for {{ $f.Name }}
 	{{ else }}
-		if v, ok := interface{}({{ accessor . }}).(interface{ Validate() error }); ok {
+		if all {
+			switch v := interface{}({{ accessor . }}).(type) {
```

Comment:
[style] Do we need the `interface{}(...)` "cast"? I believe it should work perfectly fine without it:

```suggestion
			switch v := ({{ accessor . }}).(type) {
```

Reply:
Without it the compiler throws:
```
cannot type switch on non-interface value m.GetEmbeddedExampleMessage() (type *EmbeddedExampleMessage)
```

Reply:
Thanks! TIL. I'd click resolved as you're right here, but weirdly I don't seem to be able to find the button `O_o` not sure why :/